flattener.go

```diff
@@ -58,5 +58,5 @@ type Field struct {
 	Path       []byte
 	Val        []byte
 	ArrayTrail []ArrayPos
-	IsQNumber  bool
+	isNumber   bool
```

Comment:
No change needed, just an api-design question:
Why the change to unexported? It doesn't have to be exported, but since the struct and all other fields and the previous version are all exported, it stands out.
If quamina is used as a library, it could be nice to now and it's probably more relevant for clients than e.g. ArrayTrail.

Reply:
Oh, just a mistake, should be exported.